examples/pytorch/summarization/run_summarization.py

```diff
@@ -366,6 +366,15 @@ def main():
     if model.config.decoder_start_token_id is None:
         raise ValueError("Make sure that `config.decoder_start_token_id` is correctly defined")
 
+    if (
+        hasattr(model.config, "max_position_embeddings")
+        and model.config.max_position_embeddings < data_args.max_source_length
+    ):
+        logger.info(
+            f"Increasing the model's number of position embedding vectors from {model.config.max_position_embedding} to {data_args.max_source_length}."
+        )
+        model.resize_token_embeddings(data_args.max_source_length)
```

Comment:
I feel like modifying the `max_position_embeddings` and adding some tokens should be completely opt-in by the user - I have a feeling that this could be somewhat unseen by the user but I haven't played with these scripts as much as you have

Reply:
My thinking here is that for sinusoidal positional embeddings it's 100% fine to add new tokens since they aren't randomly initialized. For learned position embeddings, it's true that it's could become a bit problematic that that weight embeddings are silently changed.

I would also be ok with adding a flag which then means that the script should crash if `max_source_length > model.config.max_position_embeddings` with an error message stating that `--resize_position_embeddings` should be passed to enable automatic resizing. On the other  we are already doing the same thing with the word embeddings here: https://github.com/huggingface/transformers/blob/fc3551a6d72e4126068aa2adf6fc8cb1cf1941ce/examples/pytorch/summarization/run_summarization.py#L364

=> so think I'd prefer to leave the script as is for consistency and because especially for seq2seq most models have 1024 seq length or have sinusoidal position embeddings like Pegasus.

Thoughts? @stas00 @patil-suraj @sgugger ?

Reply:
I see, yeah, good point! In that case, we could leave it as it is :) 

Re: `resize_token_embeddings`, these are resized using `len(tokenizer)` which only changes when the user has added new tokens, which is done outside of the script, so in this case, the user is aware that there are new tokens and token embeds will be resized, no ?

Reply:
This is a difficult question because of the word embedding just-do-it precedent. Did we have any complaints about that in the recent history?

Probably the most clean approach would be to:

1. sinusoidal - do it unconditionally
2. learned - introduce a new `--resize_position_embeddings` flag that will be tri-state 
- None (default) - do it and issue a warning that suggests to use  `--resize_position_embeddings` to explicitly control the behavior
-True: do it and issue no warning (for those of us who want to act on the warnings)
- False: throw exception as the subsequent code will fail 

I hope this proposal isn't too complicated.

Reply:
It's not trivial though to figure out if the model has sinusoidal position embeddings :-/ To me the cleanest solution is here to change the `logger.info` to `logger.warning`

Reply:
See, that's where we educate users to ignore warnings, which then defeats the purpose of warnings.

I think there needs to be a way to deal with the warning so it's no longer there should a user choose to have no warnings.

So, given your commentary of not being able to easily tell the type of the embedding, my updated proposal is then:

regardless of the type of the embedding - introduce a new `--resize_position_embeddings` flag that will be tri-state 
- None (default) - do it and issue a warning that suggests to use  `--resize_position_embeddings` to explicitly control the behavior
- True: do it and issue no warning (for those of us who want to act on the warnings)
- False:  throw exception as the subsequent code will fail 

I know I'm asking you to do more work than you believe is needed - so to support my proposition - would you like me to work on that part if others agree it's important?

I'm also fine to let it go if the others feel it's hair splitting.

Reply:
No I'm fine with your proposal and agree it's a good solution to add a flag! I'll update the PR and make it mergeable :-)